---
title: C6320 | Microsoft Docs
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
caps.latest.revision: 12
author: corob-msft
ms.author: corob
manager: jillfra
ms.openlocfilehash: 1d37a8d33f756372cd0448e94cf9f6d5e2408cf1
ms.sourcegitcommit: 68f893f6e472df46f323db34a13a7034dccad25a
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 02/15/2020
ms.locfileid: "77273422"
---
# <a name="c6320"></a>C6320
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Upozornění C6320: výraz filtru výjimky je konstanta EXCEPTION_EXECUTE_HANDLER. To může maskovat výjimky, které nebyly určeny ke zpracování.  
  
 Toto upozornění označuje vedlejší účinek použití EXCEPTION_EXECUTE_HANDLER konstanty v bloku __except. V tomto případě se příkaz v bloku \__except vždy spustí pro zpracování výjimky, včetně výjimek, které nechcete zpracovat v konkrétní funkci. Doporučujeme, abyste před manipulací výjimku kontrolovali.  
  
## <a name="example"></a>Příklad  
 Následující kód vygeneruje toto upozornění, protože __except blok se pokusí zpracovat výjimky všech typů:  
  
```  
#include <stdio.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;  // might cause access violation exception  
      // code ...  
   }   
   __except(EXCEPTION_EXECUTE_HANDLER) // warning  
   {   
      puts("in except");   
      // code ...  
   }   
}   
```  
  
 Chcete-li toto upozornění opravit, použijte `GetExceptionCode` ke kontrole konkrétní výjimky před jejím zpracováním, jak je znázorněno v následujícím kódu:  
  
```  
#include <stdio.h>   
#include <windows.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;    // might cause access violation exception   
      // code ...  
   }   
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
   {   
      puts("in except");   
      // code ...  
   }   
}  
```  
  
## <a name="see-also"></a>Viz také  
 [try-except – příkaz](https://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)
